src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -1131,7 +1131,8 @@ export class NetworkAreaDiagramViewer {
             const edgeNode: SVGGraphicsElement | null = this.container.querySelector("[id='" + edgeId + "']");
             if (edgeNode) {
                 const side = busNode.svgId == edge.busNode1 ? 0 : 1;
-                const halfEdge: HTMLElement = <HTMLElement>edgeNode.children.item(side)?.firstElementChild;
+                const childIndex = this.svgParameters.getInsertNameDesc() ? side + 1 : side;
```

Comment:
You could use the half-edge id (called angleId here, maybe we should rename it!) to query directly the half edge element.

Reply:
Used query selector instead of children list, and renamed variables